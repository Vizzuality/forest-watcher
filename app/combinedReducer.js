import * as reducers from 'redux-modules';
import { combineReducers } from 'redux';

/**
 * Separated so we can test the store state without having to mock loads of dependencies.
 */
export const combinedReducer = combineReducers(reducers);
