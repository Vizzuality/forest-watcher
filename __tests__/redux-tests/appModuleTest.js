import 'react-native';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { combinedReducer } from 'combinedReducer';
import appReducer, {
  retrySync,
  saveLastActions,
  setAppSynced,
  setCoordinatesFormat,
  setOfflineMode,
  setPristineCacheTooltip,
  showNotConnectedNotification,
  syncApp,
  updateApp
} from 'redux-modules/app';

describe('Redux App Module', () => {
// ACTION TESTS
  describe('Redux Snapshot Actions', () => {
    it('saveLastActions', () => {
      expect(saveLastActions()).toMatchSnapshot();
    });

    it('setAppSynced', () => {
      expect(setAppSynced(true)).toMatchSnapshot();
      expect(setAppSynced(false)).toMatchSnapshot();
    });

    it('setCoordinatesFormat', () => {
      expect(setCoordinatesFormat('decimal')).toMatchSnapshot();
      expect(setCoordinatesFormat('degrees')).toMatchSnapshot();
    });

    it('setOfflineMode', () => {
      expect(setOfflineMode(true)).toMatchSnapshot();
      expect(setOfflineMode(false)).toMatchSnapshot();
    });

    it('setPristineCacheTooltip', () => {
      expect(setPristineCacheTooltip(true)).toMatchSnapshot();
      expect(setPristineCacheTooltip(false)).toMatchSnapshot();
    });

    it('updateApp', () => {
      expect(updateApp()).toMatchSnapshot();
    });
  });

  describe('Redux Snapshot Thunk Actions', () => {
    let initialStoreState;
    let configuredStore;
    let store;

    beforeAll(() => {
      // create store
      initialStoreState = combinedReducer(undefined, { type: 'NONE' });
      configuredStore = configureStore([thunk]);
    });

    beforeEach(() => {
      // reset store state and clears actions
      store = configuredStore(initialStoreState);
    });

    it('retrySync', () => {
      store.dispatch(retrySync());
      expect(store.getActions()).toMatchSnapshot();
    });

    it('showNotConnectedNotification', () => {
      store.dispatch(showNotConnectedNotification());
      // todo find way to call setOfflineMode(true) then dispatch showNotConnectedNotification again
      expect(store.getActions()).toMatchSnapshot();
    });

    it('syncApp', () => {
      store.dispatch(syncApp());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

// REDUCER & STATE TESTS
  describe('Redux Reducers - Unit Tests', () => {
    it('tests', () => {
      expect(true);
    });
    // it('snapshot redux reducer initial state test', () => {
    //   expect(appReducer(undefined, { type: 'NONE' })).toMatchSnapshot();
    // });
    //
    // it('snapshot redux reducer action test', () => {
    //   expect(appReducer(undefined, setAppSynced(true))).toMatchSnapshot();
    // });
    //
    // it('snapshot redux reducer multiple action test', () => {
    //   let state = appReducer(undefined, setAppSynced(true));
    //   expect(state).toMatchSnapshot();
    //   state = appReducer(state, setOfflineMode(true));
    //   expect(state).toMatchSnapshot();
    // });
  });
});

// MOCKS
/* eslint-disable */
jest.mock('react-native-config', () => ({ API_URL: '', API_AUTH: '<TEST_API_AUTH>', MAPBOX_TOKEN: '' }));
jest.mock('react-native-cookies', () => {
  clearAll: jest.fn();
});
jest.mock('react-native-fetch-blob', () => {
  config: jest.fn();
});
jest.mock('react-native-zip-archive', () => {
  unzip: jest.fn();
});
jest.mock('Dimensions');
