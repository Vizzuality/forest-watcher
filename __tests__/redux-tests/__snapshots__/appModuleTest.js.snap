// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux App Module Redux Snapshot Actions saveLastActions 1`] = `
Object {
  "payload": undefined,
  "type": "app/SAVE_LAST_ACTIONS",
}
`;

exports[`Redux App Module Redux Snapshot Actions setAppSynced 1`] = `
Object {
  "payload": true,
  "type": "app/SET_APP_SYNCED",
}
`;

exports[`Redux App Module Redux Snapshot Actions setAppSynced 2`] = `
Object {
  "payload": false,
  "type": "app/SET_APP_SYNCED",
}
`;

exports[`Redux App Module Redux Snapshot Actions setCoordinatesFormat 1`] = `
Object {
  "payload": "decimal",
  "type": "app/SET_COORDINATES_FORMAT",
}
`;

exports[`Redux App Module Redux Snapshot Actions setCoordinatesFormat 2`] = `
Object {
  "payload": "degrees",
  "type": "app/SET_COORDINATES_FORMAT",
}
`;

exports[`Redux App Module Redux Snapshot Actions setOfflineMode 1`] = `
Object {
  "payload": true,
  "type": "app/SET_OFFLINE_MODE",
}
`;

exports[`Redux App Module Redux Snapshot Actions setOfflineMode 2`] = `
Object {
  "payload": false,
  "type": "app/SET_OFFLINE_MODE",
}
`;

exports[`Redux App Module Redux Snapshot Actions setPristineCacheTooltip 1`] = `
Object {
  "payload": true,
  "type": "app/SET_PRISTINE_CACHE_TOOLTIP",
}
`;

exports[`Redux App Module Redux Snapshot Actions setPristineCacheTooltip 2`] = `
Object {
  "payload": false,
  "type": "app/SET_PRISTINE_CACHE_TOOLTIP",
}
`;

exports[`Redux App Module Redux Snapshot Actions updateApp 1`] = `
Object {
  "type": "app/UPDATE_APP",
}
`;

exports[`Redux App Module Redux Snapshot Thunk Actions retrySync 1`] = `
Array [
  Object {
    "type": "app/RETRY_SYNC",
  },
  Object {
    "meta": Object {
      "offline": Object {
        "commit": Object {
          "type": "user/GET_USER_COMMIT",
        },
        "effect": Object {
          "url": "<TEST_API_AUTH>/user",
        },
        "rollback": Object {
          "type": "user/GET_USER_ROLLBACK",
        },
      },
    },
    "type": "user/GET_USER_REQUEST",
  },
]
`;

exports[`Redux App Module Redux Snapshot Thunk Actions showNotConnectedNotification 1`] = `
Array [
  Object {
    "type": "app/SHOW_CONNECTION_REQUIRED",
  },
]
`;

exports[`Redux App Module Redux Snapshot Thunk Actions syncApp 1`] = `
Array [
  Object {
    "meta": Object {
      "offline": Object {
        "commit": Object {
          "type": "user/GET_USER_COMMIT",
        },
        "effect": Object {
          "url": "<TEST_API_AUTH>/user",
        },
        "rollback": Object {
          "type": "user/GET_USER_ROLLBACK",
        },
      },
    },
    "type": "user/GET_USER_REQUEST",
  },
]
`;
